### PR TITLE
Press0-1119 | Press0-1141

### PR DESCRIPTION
**Pull request by mamatharao05**
[PRESS0-1119](https://jira.newfold.com/browse/PRESS0-1119) - Add / Alter visual indicator to status/date column
[PRESS0-1141](https://jira.newfold.com/browse/PRESS0-1141) -Disable most columns by default for pages and posts
This change is for Novice users only.

**Live**
Posts
image

Pages
image

**Not Live**
Posts
image

Pages
image

### CODE REVIEW

File: includes/ECommerce.php [hunk 1]
```diff
@@ -543,4 +549,81 @@ public function detect_plugin_activation( $plugin ) {
...
 	}
+
+	/**
+	 * Hide Most columns by default
+	 * Shows title and date in the page/post screen by default
+	 *
+	 * @return void
+	 */
+	public function hide_columns() {
+		if( 1 == get_option( 'onboarding_experience_level' ) ) {
```
Comment (aratidgr8) [inline includes/ECommerce.php:560]: Please make it, 
```if( get_option( 'onboarding_experience_level' )  == 1)```

File: includes/ECommerce.php [hunk 2]
```diff
@@ -543,4 +549,81 @@ public function detect_plugin_activation( $plugin ) {
...
+
+	/**
+	 * Shows status and availability under status
+	 *
+	 * @param string $column_name column names to which content needs to be updated
+	 *
+	 * @param int    $post_id Id of post/page
+	 */
+	public function custom_status_column_content( $column_name, $post_id ) {
+		if ( 'status' === $column_name ) {
```
Comment (aratidgr8) [inline includes/ECommerce.php:592]: Please make it,
``` if (  $column_name  ===  'status')```

File: includes/ECommerce.php [hunk 3]
```diff
@@ -543,4 +549,81 @@ public function detect_plugin_activation( $plugin ) {
...
+
+	/**
+	 * Add custom column header for post/page/product screen
+	 *
+	 * @param array $columns Array of column names for posts/pages
+	 */
+	public function custom_status_column( $columns ) {
+		if ( 'product' != get_post_type() && 1 == get_option( 'onboarding_experience_level' ) ) {
+			// Add 'Status' column after 'Title'
+			$columns['status'] = 'Status';
```
Comment (aratidgr8) [inline includes/ECommerce.php:579]: We need to also consider case of `pt-BR` translations

File: includes/ECommerce.php (hunk 2 above)
Comment (aratidgr8) [inline includes/ECommerce.php:592]: Also check for `pt-BR` translations

File: includes/ECommerce.php [hunk 4]
```diff
@@ -543,4 +549,81 @@ public function detect_plugin_activation( $plugin ) {
...
+			$post_status = get_post_status( $post_id );
+			// Get the post date
+			$post_date = get_post_field( 'post_date', $post_id );
+			// Get the post visibility
+			$post_visibility = get_post_field( 'post_password', $post_id );
+
+			$common_style = 'height: 24px; border-radius: 13px 13px 13px 13px;  gap: 16px; padding: 5px 10px; font-weight: 590;font-size: 12px;';
+			if ( 'publish' === $post_status ) {
+				$background_color = empty( $post_visibility ) ? '#C6E8CA' : '#FDE5CC';
+				$label_text       = empty( $post_visibility ) ? 'Published - Public' : 'Published - Password Protected';
```
Comment (aratidgr8) [inline includes/ECommerce.php:603]: Add translation tags

File: includes/ECommerce.php [hunk 5]
```diff
@@ -543,4 +549,81 @@ public function detect_plugin_activation( $plugin ) {
...
+			} else {
+				$background_color = '#E8ECF0';
+				$label_text       = $post_status;
+			}
+			// Check if coming soon option is enabled
+			$coming_soon = get_option( 'nfd_coming_soon' );
+			if ( $coming_soon ) {
+				$background_color = '#E8ECF0';
+			}
+			echo '<span style="background-color: ' . $background_color . '; ' . $common_style . '">' . $label_text . '</span><br> Last Modified: ' . mysql2date( 'Y/m/d \a\t g:i a', $post_date );
```
Comment (aratidgr8) [inline includes/ECommerce.php:616]: Add translation tags

File: includes/ECommerce.php [hunk 6]
```diff
@@ -543,4 +549,81 @@ public function detect_plugin_activation( $plugin ) {
...
+			// Get the post visibility
+			$post_visibility = get_post_field( 'post_password', $post_id );
+
+			$common_style = 'height: 24px; border-radius: 13px 13px 13px 13px;  gap: 16px; padding: 5px 10px; font-weight: 590;font-size: 12px;';
+			if ( 'publish' === $post_status ) {
+				$background_color = empty( $post_visibility ) ? '#C6E8CA' : '#FDE5CC';
+				$label_text       = empty( $post_visibility ) ? 'Published - Public' : 'Published - Password Protected';
+			} elseif ( 'private' === $post_status ) {
+				$background_color = '#CCDCF4';
+				$label_text       = 'Published - Private';
```
Comment (aratidgr8) [inline includes/ECommerce.php:606]: Add translation tags

**Review by aratidgr8 — CHANGES_REQUESTED** [review]

File: includes/ECommerce.php (hunk 2 above)
Comment (mamatharao05) [reply includes/ECommerce.php:592]: This too based on the Yoda condition check.

**Comment by mamatharao05** [comment]
@aratidgr8 Addressed the suggestions.

File: includes/ECommerce.php [hunk 7]
```diff
@@ -543,4 +549,80 @@ public function detect_plugin_activation( $plugin ) {
...
+			// Get the post visibility
+			$post_visibility = get_post_field( 'post_password', $post_id );
+
+			$common_style = 'height: 24px; border-radius: 13px 13px 13px 13px;  gap: 16px; padding: 5px 10px; font-weight: 590;font-size: 12px;';
+			if ( 'publish' === $post_status ) {
+				$background_color = empty( $post_visibility ) ? '#C6E8CA' : '#FDE5CC';
+				$label_text       = empty( $post_visibility ) ? __( 'Published - Public', 'wp-module-ecommerce' ) : __( 'Published - Password Protected', 'wp-module-ecommerce' );
+			} elseif ( 'private' === $post_status ) {
+				$background_color = '#CCDCF4';
+				$label_text       = 'Published - Private';
```
Comment (aratidgr8) [inline includes/ECommerce.php:605]: Add translation tags on line nos. 605 & 608 as well

File: includes/ECommerce.php [hunk 8]
```diff
@@ -543,4 +549,80 @@ public function detect_plugin_activation( $plugin ) {
...
+		}
+	}
+
+	/**
+	 * Add sorting for the status column
+	 *
+	 * @param array $columns Array of sortable column names for posts/pages
+	 */
+	public function sortable_columns( $columns ) {
+		$columns['status'] = 'status';
```
Comment (aratidgr8) [inline includes/ECommerce.php:625]: Add translation tag

**Review by aratidgr8 — CHANGES_REQUESTED** [review]

File: includes/ECommerce.php (hunk 8 above)
Comment (mamatharao05) [reply includes/ECommerce.php:625]: This is a column name. Cannot add the translation for this.

File: includes/ECommerce.php (hunk 2 above)
Comment (mamatharao05) [reply includes/ECommerce.php:591]: Cannot add the translation for this.

File: includes/ECommerce.php (hunk 7 above)
Comment (mamatharao05) [reply includes/ECommerce.php]: Added translation tags.

**Review by aratidgr8 — APPROVED** [review]

**Review by manikantakailasa — APPROVED** [review]

**Pull request merged**